src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.Analyzers.targets

```
@@ -13,20 +13,37 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup>
 
     <!-- Analysis level is a single property that can be used to control both the compiler warning waves
-         and enable .NET analyzers  -->
+         and enable .NET analyzers. Valid values are 'none', 'latest', 'preview', or a version number  -->
+         
+    <!-- EffectiveAnalysisLevel is used to differentiate from user specified strings (such as 'none')
+         and an implied numerical option (such as '4')-->
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'none'">4.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'latest'">5.0</EffectiveAnalysisLevel>
+    <EffectiveAnalysisLevel Condition="'$(AnalysisLevel)' == 'preview'">6.0</EffectiveAnalysisLevel>
+    
+    <!-- Automatically set AnalysisLevel if not specified -->
     <AnalysisLevel Condition="'$(AnalysisLevel)' == '' And
                               '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And
                               $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0'))">5.0</AnalysisLevel>
+    <!-- Set EffectiveAnalysisLevel to the value of AnalysisLevel if it is a version number -->
+    <EffectiveAnalysisLevel Condition="'$(EffectiveAnalysisLevel)' == '' And 
+                                       '$(AnalysisLevel)' != ''">$(AnalysisLevel)</EffectiveAnalysisLevel>
 
     <!-- EnableNETAnalyzers Allows analyzes to be disabled in bulk via msbuild if the user wants to -->
     <EnableNETAnalyzers Condition="'$(EnableNETAnalyzers)' == '' And 
-                                   '$(AnalysisLevel)' != '' And
-                                    $([MSBuild]::VersionGreaterThanOrEquals($(AnalysisLevel), '5.0'))">true</EnableNETAnalyzers>
+                                   '$(EffectiveAnalysisLevel)' != '' And
+                                    $([MSBuild]::VersionGreaterThanOrEquals($(EffectiveAnalysisLevel), '5.0'))">true</EnableNETAnalyzers>
     <EnableNETAnalyzers Condition="'$(EnableNETAnalyzers)' == ''">false</EnableNETAnalyzers>
     
-    <!-- Compiler warning level, defaulted to 4. We promote it to 5 if the user has set analysis level to 5 or higher -->
-    <WarningLevel Condition="'$(AnalysisLevel)' != '' And
-                              $([MSBuild]::VersionGreaterThanOrEquals($(AnalysisLevel), '5.0'))">5</WarningLevel>
+    <!-- Compiler warning level, defaulted to 4. We promote it to 5 if the user has set analysis level to 5 or higher
+         NOTE: at this time only the C# compiler supports warning waves -->
+    <WarningLevel Condition="'$(Language)' == 'C#' And
+                             '$(EffectiveAnalysisLevel)' != '' And
+                             $([MSBuild]::VersionGreaterThanOrEquals($(EffectiveAnalysisLevel), '5.0'))">5</WarningLevel>
```

Shouldn't this default to `EffectiveAnalysisLevel` so we have never have to change this in future?
```suggestion
                             $([MSBuild]::VersionGreaterThanOrEquals($(EffectiveAnalysisLevel), '5.0'))">$(EffectiveAnalysisLevel)</WarningLevel>
```

---

it must be an integer. It cannot be a string like '5.0'. I must be '5'

```cmd
> csc -warn:5.0 Hello.cs
Microsoft (R) Visual C# Compiler version 3.8.0-2.20410.4 (2c664a69)
Copyright (C) Microsoft Corporation. All rights reserved.

error CS2035: Command-line syntax error: Missing ':<number>' for 'warn' option
```